Dockerfile

```diff
@@ -0,0 +1,19 @@
+# Stage 1: Install build tools and dependencies
+FROM python:3.11.5-slim-bookworm as base
+WORKDIR /app
+RUN pip install poetry
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-dev
+
+# Stage 2: Copy application code
```

Comment:
"Stage" here might be slightly misleading wording, given that it's not a separate Dockerfile stage, but I appreciate it's an overloaded word!